python/ray/serve/tests/test_batching.py

```diff
@@ -214,6 +215,36 @@ async def call(arg):
         await t3
 
 
+@pytest.mark.asyncio
+async def test_batch_timeout_empty_queue():
+    """Check that Serve waits for before creating the first batch."""
+
+    @serve.batch(max_batch_size=10, batch_wait_timeout_s=0.1)
+    async def no_op(requests):
+        return ["No-op"] * len(requests)
+
+    # Run one request, so internal batch queue gets constructed
+    await get_or_create_event_loop().create_task(no_op(None))
+
+    time.sleep(0.1)
+
+    tasks = [get_or_create_event_loop().create_task(no_op(None)) for _ in range(9)]
+    done, _ = await asyncio.wait(tasks, timeout=0.01)
+
+    # Due to the long timeout, none of the tasks should finish until a tenth
+    # request is submitted
+    assert len(done) == 0
+
+    tasks.append(get_or_create_event_loop().create_task(no_op(None)))
+    done, _ = await asyncio.wait(tasks, timeout=0.01)
+
+    # All the timeout tasks should be finished
+    assert len(done) == 10
+    for task in tasks:
+        assert task in done
+        assert await task == "No-op"
```

Comment:
maybe better way to write it depending on your preference :)
```suggestion
    assert set(tasks) == set(done)
    assert all(t.result() == "No-op" for t in tasks)
```

Reply:
Cool, I didn't know tasks were hashable.